### ISSUE TITLE
model1.par = model2.par has unexpected results

### ISSUE THREAD

**Issue by dlakaplan** [nanograv/PINT#1809]
From @scottransom :
```
from io import StringIO
import pint.models as pm
import pint.fitter as pf

par1 = """
RAJ      17:48:04.53480592  0          0.00005548
DECJ     -24:46:34.6586741  0           0.0279874
F0              118.538254  1  0.0000000000054826
F1            8.485977D-15  0  9.543894501911D-20
PEPOCH        53000.000000
DM              237.062679  0            0.005850
SOLARN0              10.00
EPHEM               DE421
CLK                 UNCORR                          
UNITS               TDB
"""

par2 = """
RAJ      17:48:04.53480592  0          0.00005548
DECJ     -24:46:34.6586741  0           0.0279874
F0              666.666666  1  0.0000000000054826
F1            8.485977D-15  0  9.543894501911D-20
PEPOCH        53000.000000
DM              237.062679  0            0.005850
SOLARN0              10.00
EPHEM               DE421
CLK                 UNCORR                          
UNITS               TDB
"""

m1 = pm.get_model(StringIO(par1))
m2 = pm.get_model(StringIO(par2))
print(m1)
# The following works if you use this instead:
# m1.F0.value = m2.F0.value
m1.F0 = m2.F0
# Following should(?) be different, but it isn't
print(m1)

# These show that the F0s are both consistent
print(m1.F0)
print(m2.F0)
assert(m1.F0==m2.F0)
```

**Comment by dlakaplan** [comment]
When the above is run, it looks like the assignment into `m1` is not part of the correct component.  so rather than copying by reference or value it's doing a third even weirder thing

**Comment by dlakaplan** [comment]
`model1.par1 = model1.par2` is explicitly allowed, and copies `.quantity`, `.uncertainty`, and `.frozen` attributes.  But it needs to have the two parameters have different names in order to work

**Comment by dlakaplan** [comment]
The rationale for that exclusion is that it would otherwise override the name of `par1`.  That usage was for things like:
```
m.POSEPOCH = m.PEPOCH
```

**Comment by scottransom** [comment]
Could you still get that same behavior above (i.e. copying if the names are different) by simply checking that the names are different first, and if they are the same then give a reference?  As mentioned in the PINT Slack channel, I think assigning by reference should be the default since that is what Python does by default.

**Comment by dlakaplan** [comment]
Yes, I think we could.  But I worry about how that would interact with things like fitters that are varying parameters (not relevant for `PEPOCH`, but relevant for `F0`).

I think this could use a bit of thought about the different cases and expected behaviors.  Maybe for discussion on the next call?

**Comment by dlakaplan** [comment]
Cases to consider:
* `m1.p1 = m1.p2`
* `m1.p1 = m2.p1`
* `m1.comp1 = m2.comp1`
* should those all raise exceptions, or if not what behavior to implement (copy via reference or values)?
* cases where `p1` is fittable vs not
* do we want to allow assignment via reference at all?  If we override it to do assignment via value, should we have another way to do via reference (which is python default)?

**Comment by dlakaplan** [comment]
If I do:
`m1.PSR.value = m2.PSR.value`
it appears that `id(m1.PSR.value)` is the same as `id(m2.PSR.value)`, but that may just be because they are strings.  If I then change `m2.PSR.value` `m1.PSR.value` will not change.

**Comment by dlakaplan** [comment]
If I do:
```
x=10*u.m
y=20*u.m
x=y
y=30*u.m
print(x)
```
returns 20m, so when assigning Quantities it appears to pass via value.

**Comment by dlakaplan** [comment]
If I do:
```
m1.components['Spindown'].F0=m2.components['Spindown'].F0
```
and then
```
m2.F0.value=10
```
then `m1.F0` is 10.

**Comment by dlakaplan** [comment]
So the consensus is to make:
```
m1.F0 = m2.F0
```
behave like:
```
m1.components['Spindown'].F0=m2.components['Spindown'].F0
```

**Comment by dlakaplan** [comment]
But also allow:
```
m1.PSR = m2.PSR
```
since those are top-level.

**Comment by dlakaplan** [comment]
Closed by #1810
